stdlib/public/core/StringBridge.swift

```diff
@@ -661,6 +661,18 @@ internal func _SwiftCreateBridgedString_DoNotCall(
   return Unmanaged<AnyObject>.passRetained(str._bridgeToObjectiveCImpl())
 }
 
+@available(SwiftStdlib 6.1, *)
+@_spi(Foundation) public func _SwiftCreateImmortalString_ForFoundation(
+  buffer: UnsafeBufferPointer<UInt8>,
+  isASCII: Bool
```

Comment:
In the current implementation the `isASCII` flag here is redundant, but if we later choose to accept that Foundation is doing the validation for us, having it will allow us to remove the `validateUTF8` call without losing the ASCII flag.

Reply:
The forward-looking plan for including the parameter makes sense, but if we're not presently trusting Foundation to provide valid ASCII, we probably also shouldn't be respecting this flag, right?

`isASCII || extraInfo.isASCII` would be true if we're actually passed valid non-ASCII UTF-8.

Reply:
I suppose that's true, yeah. Good point.